subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppLibraryPublishingIntegrationTest.groovy

```diff
@@ -77,58 +80,70 @@ class CppLibraryPublishingIntegrationTest extends AbstractCppInstalledToolChainI
-        debug.assertPublished()
-        debug.assertArtifactsPublished(withSharedLibrarySuffix("test_debug-1.2"), withLinkLibrarySuffix("test_debug-1.2"), "test_debug-1.2.pom", "test_debug-1.2-module.json")
-        debug.artifactFile(type: sharedLibraryExtension).assertIsCopyOf(sharedLibrary("build/lib/main/debug/test").file)
-        debug.artifactFile(type: linkLibrarySuffix).assertIsCopyOf(sharedLibrary("build/lib/main/debug/test").linkFile)
-
-        debug.parsedPom.scopes.isEmpty()
-
-        def debugMetadata = debug.parsedModuleMetadata
-        debugMetadata.variants.size() == 2
-        def debugLink = debugMetadata.variant('debug-link')
-        debugLink.dependencies.empty
-        debugLink.files.size() == 1
-        debugLink.files[0].name == linkLibraryName('test')
-        debugLink.files[0].url == withLinkLibrarySuffix("test_debug-1.2")
-        def debugRuntime = debugMetadata.variant('debug-runtime')
-        debugRuntime.dependencies.empty
-        debugRuntime.files.size() == 1
-        debugRuntime.files[0].name == sharedLibraryName('test')
-        debugRuntime.files[0].url == withSharedLibrarySuffix("test_debug-1.2")
-
-        def release = repo.module('some.group', 'test_release', '1.2')
-        release.assertPublished()
-        release.assertArtifactsPublished(withSharedLibrarySuffix("test_release-1.2"), withLinkLibrarySuffix("test_release-1.2"), "test_release-1.2.pom", "test_release-1.2-module.json")
-        release.artifactFile(type: sharedLibraryExtension).assertIsCopyOf(sharedLibrary("build/lib/main/release/test").file)
-        release.artifactFile(type: linkLibrarySuffix).assertIsCopyOf(sharedLibrary("build/lib/main/release/test").linkFile)
-
-        release.parsedPom.scopes.isEmpty()
-
-        def releaseMetadata = release.parsedModuleMetadata
-        releaseMetadata.variants.size() == 2
-        def releaseLink = releaseMetadata.variant('release-link')
-        releaseLink.dependencies.empty
-        releaseLink.files.size() == 1
-        releaseLink.files[0].name == linkLibraryName('test')
-        releaseLink.files[0].url == withLinkLibrarySuffix("test_release-1.2")
-        def releaseRuntime = releaseMetadata.variant('release-runtime')
-        releaseRuntime.dependencies.empty
-        releaseRuntime.files.size() == 1
-        releaseRuntime.files[0].name == sharedLibraryName('test')
-        releaseRuntime.files[0].url == withSharedLibrarySuffix("test_release-1.2")
+        mainMetadata.variant("debugShared-link").availableAt.coords == "some.group:test_debugShared:1.2"
+        mainMetadata.variant("debugShared-runtime").availableAt.coords == "some.group:test_debugShared:1.2"
+        mainMetadata.variant("releaseShared-link").availableAt.coords == "some.group:test_releaseShared:1.2"
+        mainMetadata.variant("releaseShared-runtime").availableAt.coords == "some.group:test_releaseShared:1.2"
+
+        assertSharedLibraryPublished(repo, 'debug')
```

Comment:
we don't want more modules, I think. The shared and static debug binaries should live in the debug module, and the shared and static release binaries should live in the release module.